### DISCUSSION TITLE
Supporting direct display output via VK_KHR_display

### DISCUSSION THREAD

**Discussion by theoparis** [gfx-rs/wgpu#5949] [Ideas]
It would be nice to see some kind of interface for displays similar to https://github.com/gfx-rs/gfx/pull/3759 from years ago... Not sure why this didn't get ported over.

Alternatively I am considering just going with vulkanalia and writing an entire vulkan renderer from scratch, however the last time I tried that it just segfaulted 😕

**Comment by 0x000C0A71** [comment]
I second this!
Maybe just a `WGPUSurfaceDescriptorFromGBMSurface` that is able to target a GBM surface, as to not make this dependent on using the vulkan backend. Because, from what I've seen, it seems to be outright impossible to render directly to a display without sharing a buffer with another context (e.g. a separate vulkan instance) dedicated to copying the data to a DRM/GBM frame buffer.
Maybe this could also be required for VR via DRM leasing.
